cmd/podman/pod_ps.go

```diff
@@ -0,0 +1,574 @@
+	switch filter {
+	case "ctr-ids":
+		return func(p *libpod.Pod) bool {
+			ctrIds, err := runtime.ContainerIDsInPod(p)
+			if err != nil {
+				return false
+			}
+			return util.StringInSlice(filterValue, ctrIds)
+		}, nil
+	case "ctr-names":
+		return func(p *libpod.Pod) bool {
+			ctrNames, err := runtime.ContainerNamesInPod(p)
+			if err != nil {
+				return false
+			}
+			return util.StringInSlice(filterValue, ctrNames)
+		}, nil
+	case "ctr-number":
+		return func(p *libpod.Pod) bool {
+			ctrIds, err := runtime.ContainerIDsInPod(p)
+			if err != nil {
+				return false
+			}
+
+			fVint, err2 := strconv.Atoi(filterValue)
+			if err2 != nil {
+				return false
+			}
+			return len(ctrIds) == fVint
+		}, nil
+	case "ctr-status":
+		if !util.StringInSlice(filterValue, []string{"created", "restarting", "running", "paused", "exited", "unknown"}) {
+			return nil, errors.Errorf("%s is not a valid status", filterValue)
+		}
+		return func(p *libpod.Pod) bool {
+			ctrs, err := runtime.ContainersInPod(p)
+			if err != nil {
+				return false
+			}
+			for _, ctr := range ctrs {
+				status, err := ctr.State()
+				if err != nil {
+					return false
+				}
+				state := status.String()
+				if status == libpod.ContainerStateConfigured {
+					state = "created"
+				}
+				if state == filterValue {
+					return true
+				}
+			}
+			return false
+		}, nil
+	case "id":
+		return func(p *libpod.Pod) bool {
+			return strings.Contains(p.ID(), filterValue)
+		}, nil
+	case "label":
+		return func(p *libpod.Pod) bool {
+			for _, label := range p.Labels() {
+				if label == filterValue {
+					return true
+				}
+			}
+			return false
+		}, nil
+	case "name":
+		return func(p *libpod.Pod) bool {
+			return strings.Contains(p.Name(), filterValue)
+		}, nil
+	case "status":
+		if !util.StringInSlice(filterValue, []string{"stopped", "running", "paused", "exited", "dead", "created"}) {
+			return nil, errors.Errorf("%s is not a valid pod status", filterValue)
+		}
+		return func(p *libpod.Pod) bool {
+			ctrs, err := runtime.ContainersInPod(p)
+			if err != nil {
+				return false
+			}
+			status, err := getPodStatus(ctrs)
+			if err != nil {
+				return false
+			}
+			if strings.ToLower(status) == filterValue {
+				return true
+			}
+			return false
+		}, nil
+	}
+	return nil, errors.Errorf("%s is an invalid filter", filter)
+}
+
+// generate the template based on conditions given
+func genPodPsFormat(format string, quiet, cgroup, labels, names, ids, status bool) string {
+	if format != "" {
+		// "\t" from the command line is not being recognized as a tab
+		// replacing the string "\t" to a tab character if the user passes in "\t"
+		return strings.Replace(format, `\t`, "\t", -1)
+	}
+	if quiet {
+		return formats.IDString
+	}
+	format = "table {{.ID}}\t{{.Name}}\t{{.Status}}"
+	if names || ids || status {
+		format += "\t{{.ContainerInfo}}"
```

Comment:
Do we want this in the middle? This is probably the largest column. Might be better to stick it at the far end so it has room to grow on large terminals